### PR TITLE
Remove use of deprecated macros

### PR DESCRIPTION
**Pull request by oldnewthing**

### CODE REVIEW

File: include/wil/registry.h [hunk 1]
```diff
@@ -49,7 +49,7 @@ namespace wil
     //     {
     //         // invalidate cached registry data here
     //     });
-    // RETURN_HR_IF_FALSE(E_OUTOFMEMORY, watcher);
+    // RETURN_HR_IF(E_OUTOFMEMORY, !watcher);
```
Comment (dunhor) [inline include/wil/registry.h]: `RETURN_IF_NULL_ALLOC` perhaps seems better for this one

Comment (oldnewthing) [reply include/wil/registry.h]: Doesn't work because `RETURN_IF_NULL_ALLOC` requires a pointer.

Comment (dunhor) [reply include/wil/registry.h]: Seems to work for me. I think it just requires something that's convertible to `bool`

Comment (oldnewthing) [reply include/wil/registry.h]: It wants something [convertible to a pointer](https://github.com/microsoft/wil/blob/6112ec16d8a09402ef980c5b951e17bda2e5898c/include/wil/result_macros.h#L607), but maybe there's a secret converter inside the watcher.

Comment (dunhor) [reply include/wil/registry.h]: Ahh, okay. `unique_registry_watcher_nothrow` is a `unique_any_t`, which [is comparable with `nullptr_t`](https://github.com/microsoft/wil/blob/6112ec16d8a09402ef980c5b951e17bda2e5898c/include/wil/resource.h#L340)

Comment (oldnewthing) [reply include/wil/registry.h]: Mystery solved. (I was worried that we constructed a temporary `unique_event_watcher_nothrow(nullptr)`, did the comparison, and then destructed the temporary.) Should I use `RETURN_IF_NULL_ALLOC`?

**Pull request merged**